netutils/netinit/netinit.c

```diff
@@ -358,6 +448,13 @@ static void netinit_set_macaddr(void)
     defined(CONFIG_NET_IEEE802154)
 static void netinit_set_ipaddrs(void)
 {
+#ifdef CONFIG_FSUTILS_IPCFG
+  if (netinit_get_ipaddrs() == OK)
+    {
+      return;
+    }
+
+#endif
```

Comment:
It is a C90 violation to place executable logic before data definitions.

Reply:
That should be easy to fix, in the re-write.